src/pages/User.jsx

```diff
@@ -51,18 +51,39 @@ const User = () => {
                     const followUserId = res.data.users.find((user) => user.username === username)._id
 
                     let url
-                    url=isFollowing?`/api/users/unfollow/${followUserId}/`:`/api/users/follow/${followUserId}/`
+                    url = isFollowing ? `/api/users/unfollow/${followUserId}/` : `/api/users/follow/${followUserId}/`
                     const res1 = await axios({
                         method: 'post',
                         url: url,
                         headers: {
                             authorization: localStorage.getItem('token')
                         }
                     })
-                    isFollowing?setIsFollowing(false):setIsFollowing(true)
-                }}>{isFollowing?'Unfollow':'Follow'}</button>
+
+                    if (isFollowing)
+                        setIsFollowing(false);
+
+                    else {
+                        setIsFollowing(true);
+                        
+                        (async () => {
+                            const token = localStorage.getItem('token')
+                            const base64Url = token.split('.')[1];
+                            const base64 = base64Url.replace('-', '+').replace('_', '/');
+                            const selfname=JSON.parse(window.atob(base64)).username;
+                
+                            const res=await axios({
+                                method:'get',
+                                url:`/api/users`
+                            })
+                
+                            const self=res.data.users.find((user)=>user.username===selfname)
```

Comment:
Suggestion: these functions can be extracted in separated function then use that function here